File: rust/problem_09.rs
// Copyright 2014 Mitchell Kember. Subject to the MIT License.
// Project Euler: Problem 9
// Special Pythagorean triplet

fn attempt() -> Option<int> {
	for a in range(1, 500) {
		for b in range(a + 1, (1000 - a)/2) {
			let c = 1000 - a - b;
			if a * a + b * b == c * c {
				return Some(a * b * c);
			}
		}
	}
	None
}

pub fn solve() -> int {
	attempt().unwrap()
}

File: c++/problem_09.cpp
// Copyright 2016 Mitchell Kember. Subject to the MIT License.
// Project Euler: Problem 9
// Special Pythagorean triplet

namespace problem_09 {

long solve() {
	long i = 0, j = 0, k = 0;
	for (i = 1; i < 333; ++i) {
		for (j = i + 1; j < (1000 - i)/2; ++j) {
			k = 1000 - i - j;
			if (i*i + j*j == k*k) {
				goto end;
			}
		}
	}
end:
	return i * j * k;
}

} // namespace problem_09
